### PR TITLE
Revamp gui

### PR DESCRIPTION
**Pull request by pythcoiner**
This PR introduce a new structure of the GUI:
- a85663a98289337c653245b7d565c891b18f412b  split bitcoin_app() in 2 function: get_latests_app() and bitcoin_latest_app(), so the GUI can get both version in a single API call
- 805d7b841f80f4e3a6b60f6b8499930d46c29380 fix an error issued if we want to list apps and no app installed on device.
- 64e8af4135fc4b5d829ae13df3c1eb7fa9e81a4a introduce the new GUI structure itself, the genuine check is now doable only once per launch and add the update feature.

image

### CODE REVIEW

**Comment by jp1ac4** [comment]
This is looking good. I did some quick testing with a Ledger Nano X, firmware v2.0.1 and with Bitcoin Test app v2.0.1 installed.

When I first open the GUI, it gives me the option to update the Bitcoin Test app:

image

After running the update, it still shows the option to update. If I reopen the GUI without disconnecting the Nano X, it then shows I have the latest app:

image

I was able to successfully install the Bitcoin app. Again, it still gave the update option after installing, but when I reopened it was all good:

image

The genuine check works fine:

image

**Comment by pythcoiner** [comment]
Screenshot looks like the window is very large, right? (Should be 500 x 450 fixed size)

**Comment by pythcoiner** [comment]
It seems the icon in install/update button does not appear properly?

**Comment by pythcoiner** [comment]
I'll check why the state of button is not updated after update

**Comment by pythcoiner** [comment]
> After running the update, it still shows the option to update.

i just try to reproduce it but cannot

**Comment by jp1ac4** [comment]
> Screenshot looks like the window is very large, right? (Should be 500 x 450 fixed size)

It opens small by default, but I was able to maximise it still to the whole screen.

> It seems the icon in install/update button does not appear properly?

That's right, it just shows an empty rectangle.

> i just try to reproduce it but cannot

I'm able to reproduce it only if I open the GUI before connecting my Ledger device (while GUI is displaying "Please connect a device and unlock it...").

**Comment by pythcoiner** [comment]
> It opens small by default, but I was able to maximise it still to the whole screen.

Which Desktop environment are you using?

**Comment by pythcoiner** [comment]
> I'm able to reproduce it only if I open the GUI before connecting my Ledger device (while GUI is displaying "Please connect a device and unlock it...").

Ok, this should be fixed now!

**Comment by pythcoiner** [comment]
> That's right, it just shows an empty rectangle.
> 

humm, does the icons of Liana Left Menu appear well on your side ?

image

**Comment by jp1ac4** [comment]
> Which Desktop environment are you using?

It's Ubuntu 22.04.

> Ok, this should be fixed now!

Great, I tested this and it's working now: When I connect my Ledger, the already-installed app is shown as "Latest" and after installing an app it's also shown as "Latest".

> humm, does the icons of Liana Left Menu appear well on your side ?

Yep, the Liana GUI icons are appearing correctly.

**Comment by darosior** [comment]
Looks great. It would be nice to have a visual hint when the user needs to confirm an action on his device. The issue is it's not always the case. Maybe we could only have it when listing the applications.

One alternative would be to have a load spinner for the values which aren't set. The user would figure out something's pending and check his device. Does that sound doable? (probably in followup work)

**Comment by pythcoiner** [comment]
> It would be nice to have a visual hint when the user needs to confirm an action on his device. The issue is it's not always the case. Maybe we could only have it when listing the applications.

i think we have a message at the bottom for each cases the user need to interract:
- accept for list applications (device displays "Allow listing applications")
image


- accept for install apps (early before the device displays "Allow ledger manager")
image

( for this one we send the message early, as if we want to hint user at same time message appear on device screen, we should pass a callback to `lib.rs::install_bitcoin_app()` )

**Comment by pythcoiner** [comment]
> One alternative would be to have a load spinner for the values which aren't set. The user would figure out something's pending and check his device. Does that sound doable? (probably in followup work)

Yes i think we can have a `bubling sardine` in a follow up

File: gui/src/gui.rs [hunk 1]
```diff
@@ -298,47 +283,237 @@ impl Application for LedgerInstaller {
...
+    test_latest: Version,
+    device_busy: bool,
+) -> Container<'a, Message, Theme, Renderer> {
+    let network_size = 25;
+    let version_color = theme::color::GREY_3;
+    let vertical_rule_position = 230;
+
+    // It looks weird that we load iconex-icons.ttf by its name: Untitled1
+    const ICONEX_ICONS: Font = Font::with_name("Untitled1");
+
```
Comment (pythcoiner) [inline gui/src/gui.rs:368]: It's not clear to me how Iced handle the font, i guess `Untitled1` is the inner name of the font (that seems not set...) in `iconex-icons.ttf`, but i'm not sure if the font is loaded in our binary or call by OS....

**Comment by pythcoiner** [comment]
> > Which Desktop environment are you using?
> 
> It's Ubuntu 22.04.
> 
> > Ok, this should be fixed now!
> 
> Great, I tested this and it's working now: When I connect my Ledger, the already-installed app is shown as "Latest" and after installing an app it's also shown as "Latest".
> 
> > humm, does the icons of Liana Left Menu appear well on your side ?
> 
> Yep, the Liana GUI icons are appearing correctly.

We now load fonts into the binary, can you try if it fix your display issue?

**Comment by jp1ac4** [comment]
> We now load fonts into the binary, can you try if it fix your display issue?

That's working now thanks:
image

**Comment by darosior** [comment]
Did some basic sanity check, working well.

**Pull request merged**